### DISCUSSION TITLE
Is there a way to force the redirect uri to be https ?

### DISCUSSION THREAD

**Discussion by brahma-dev** [Dan6erbond/sk-auth#35] [Q&A] [answered]
I have setup sveltekit behind nginx, which also handles https. 
The redirect uri generated though is http only. How can I force it to be https ?

**Comment by Dan6erbond** [comment, marked as answer]
Hey there, you can use the `redirect` callback as seen in the README or better, specify the `host` in the config when constructing `SvelteKitAuth`. A reference for `AuthConfig` in [`auth.ts`](https://www.github.com/Dan6erbond/sk-auth/tree/main/src%2Fauth.ts).

**Reply by brahma-dev** [reply]
Thanks. I tried this to validate but it appears it's not being called.
```
redirect: (url) => {
	console.log(url);
	return url.replace('http://','https://');
}
```

**Reply by Dan6erbond** [reply]
Hmm. Interesting. That means there's a bug somewhere in the flow. Do you mind creating an issue on this and mention the provider you use?

As for using `https:` for redirects - did setting the `host` in the config work for this purpose?

**Reply by brahma-dev** [reply]
It appears that the root of the problem is that svelte doesn't provide protocol: https://github.com/sveltejs/kit/issues/1240 / https://github.com/sveltejs/kit/issues/1241
So I'll wait until they fix that. 

For now I am patching up sk-auth in postinstall 
    ```"postinstall": "sed -i 's/http:\\/\\/\\${host}/https:\\/\\/\\${host}/' ./node_modules/sk-auth/dist/auth.js"```

**Reply by Dan6erbond** [reply]
Well, the `redirect()` callback should definitely work, so we'll be looking into that, and I already created an issue. See #39.

On top of that our config provides the `host` option as you can see here. So you shouldn't have to patch the library at all:

```ts
interface AuthConfig {
  providers: Provider[];
  callbacks?: AuthCallbacks;
  jwtSecret?: string;
  jwtExpiresIn?: string | number;
  host?: string;
  basePath?: string;
}
```

That config option is used by `getBaseUrl()` in [`auth.ts`](https://github.com/Dan6erbond/sk-auth/blob/main/src/auth.ts#L72) which means you definitely won't have to patch up anything. In fact, you're patching that precise line where you could just substitute it with the usage of that config value, allowing SvelteKitAuth to still work in development with HTTP and switch to HTTPS in production using environment variables.

**Reply by brahma-dev** [reply]
The issue is that I am using the host header to get the hostname. I can't have it in the environment variable. The only thing I can ensure is that it will always be https.

The patch solves it for now. Whenever svelte kit starts providing the protocol we can use that.

**Reply by Dan6erbond** [reply]
Ah, I see. That's an interesting use-case. In that case, stay subscribed to #39 so you can use the `redirect()` callback once we get that working again!
